ecms_base/modules/custom/ecms_blocks/src/Plugin/Block/PromotionsNodeSpecificBlock.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\ecms_blocks\Plugin\Block;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Session\AccountInterface;
+
+/**
+ * Provides a listing of promos referenced by the node.
+ *
+ * @Block(
+ *   id = "ecms_promotions_node_specific",
+ *   admin_label = @Translation("Promotions - Node Specific"),
+ *   context_definitions = {
+ *     "node" = @ContextDefinition("entity:node", label = @Translation("Node"))
+ *   }
+ * )
+ */
+class PromotionsNodeSpecificBlock extends BlockBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build(): array {
+    // Load node from context.
+    $node = $this->getContextValue('node');
+
+    // Assume what the promotions field machine name is.
+    $field_string_guess = 'field_' . $node->bundle() . '_promotions';
+
+    // Using a switch in case an entity doesn't follow the above pattern.
+    // Set $promos to an array of promotion entities.
+    switch (TRUE) {
+      case ($node->hasField($field_string_guess)):
+        $promos = $node->get($field_string_guess)->referencedEntities();
+        break;
+
+      default:
+        $promos = NULL;
+    }
+
+    // If there are promos, render their teaser view.
+    if (count($promos) > 0) {
+      // Get language code.
+      $language = \Drupal::languageManager()->getCurrentLanguage()->getId();
+
+      // Return a list of rendered teaser nodes.
+      $builder = \Drupal::entityTypeManager()->getViewBuilder('node');
```

Comment:
We should use service injection for the language manager and the entity type manager. Changing the class definition to this:
```
class TestBlock extends BlockBase implements ContainerFactoryPluginInterface {

  /**
   * The language_manager service.
   *
   * @var \Drupal\Core\Language\LanguageManagerInterface
   */
  protected $languageManager;

  /**
   * The entity_type.manager service.
   *
   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
   */
  protected $entityTypeManager;

  /**
   * {@inheritDoc}
   */
  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
    return new static(
      $configuration,
      $plugin_id,
      $plugin_definition,
      $container->get('language_manager'),
      $container->get('entity_type.manager')
    );
  }

  /**
   * TestBlock constructor.
   *
   * @param array $configuration
   *   Configuration array for the block.
   * @param $plugin_id
   *   The plugin id.
   * @param $plugin_definition
   *   The plugin definition.
   * @param \Drupal\Core\Language\LanguageManagerInterface $languageManager
   *   The language_manager service.
   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
   *   The entity_type.manager service.
   */
  public function __construct(array $configuration, $plugin_id, $plugin_definition, LanguageManagerInterface $languageManager, EntityTypeManagerInterface $entityTypeManager) {
    parent::__construct($configuration, $plugin_id, $plugin_definition);

    $this->languageManager = $languageManager;
    $this->entityTypeManager = $entityTypeManager;
  }
```

Then, `\Drupal::languageManager()` becomes `$this->languageManager` and `\Drupal::entityTypeManager()` becomes `$this->entityTypeManager`.